awspricing/offers.py

```diff
@@ -114,7 +117,8 @@ def _generate_reverse_sku_mapping(self,
             if (product_family is not None and
                     product['productFamily'] != product_family):
                 continue
-            attrs = [product['attributes'][attr] for attr in attribute_names]
+            attrs = [product['attributes'][attr]
+                     for attr in attribute_names if attr in product['attributes']]
```

Comment:
I added the if conditional here to work around the **database_edition** parameter required to avoid hash collisions; but not always existing for some database engines.